pkg/apis/serving/v1beta1/inference_service_status.go

```diff
@@ -105,6 +111,96 @@ const (
 	IngressReady apis.ConditionType = "IngressReady"
 )
 
+type ModelStatus struct {
+	// Whether the available predictor endpoint reflects the current Spec or is in transition
+	// +kubebuilder:default=UpToDate
+	TransitionStatus TransitionStatus `json:"transitionStatus"`
+
+	// High level state string: Pending, Standby, Loading, Loaded, FailedToLoad
+	// +kubebuilder:default=Pending
+	ActiveModelState ModelState `json:"activeModelState"`
+	// +kubebuilder:default=""
+	TargetModelState ModelState `json:"targetModelState"`
+
+	// Details of last failure, when load of target model is failed or blocked
+	//+optional
+	LastFailureInfo *FailureInfo `json:"lastFailureInfo,omitempty"`
+
+	// How many copies of this predictor's models failed to load recently
+	// +kubebuilder:default=0
+	FailedCopies int `json:"failedCopies"`
+}
+
+// TransitionStatus enum
+// +kubebuilder:validation:Enum=UpToDate;InProgress;BlockedByFailedLoad;InvalidSpec
+type TransitionStatus string
+
+// TransitionStatus Enum values
+const (
+	// Predictor is up-to-date (reflects current spec)
+	UpToDate TransitionStatus = "UpToDate"
+	// Waiting for target model to reach state of active model
+	InProgress TransitionStatus = "InProgress"
+	// Target model failed to load
+	BlockedByFailedLoad TransitionStatus = "BlockedByFailedLoad"
+	// TBD
+	InvalidSpec TransitionStatus = "InvalidSpec"
```

Comment:
What are the cases for invalid spec?

Reply:
@yuzisun Can we update InvalidSpec status for NoMatching/Disabled/Invalid Runtime provided?

Reply:
In MM, this is for when the predictor's currently active model configuration was not transitioned to match its current spec because the current spec is invalid. This would typically be coupled with an error message in the lastFailureInfo object.